src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -143,6 +156,14 @@ assign(ReactDOMTextComponent.prototype, {
   },
 
   unmountComponent: function() {
+    var openingComment = this._openingComment;
```

Comment:
So this and line 147 are the only places that need to reference `this._openingComment` now. I made `unmountComponent` remove the opening comment + text nodes here but the closing comment node is still removed via ReactMultiChild -> DOMChildrenOperations, since it's what's returned by `getNativeNode`. I didn't need to change anything about ReactMultiChild or `DOMChildrenOperations.processUpdates` that way.